diff --git a/app/nodejs/server.js b/app/nodejs/server.js
index fef0510..2f18a77 100755
--- a/app/nodejs/server.js
+++ b/app/nodejs/server.js
@@ -7,6 +7,20 @@
 const express = require("express");
 const session = require("express-session");
 const path = require("path");
+const redis = require("redis");
+
+/**
+ * External services configuration
+ */
+
+redis_client = redis.createClient(6379,"redis",{ "password": "foobared"});
+redis_client.on("error", function (err) {
+  console.log("Error " + err);
+});
+
+redis_client.on("ready", function (err) {
+  redis_client.set("masuno", 0);
+});
 
 /**
  * App Variables
@@ -44,6 +58,14 @@ app.get("/saludos", (req, res) => {
 /**
  * Rutas para el factor4
  */
+ 
+app.get("/masuno", (req, res) => {
+  redis_client.get("masuno", function(err, masuno) {
+    masuno++
+    redis_client.set("masuno", masuno);
+    res.status(200).send("Contador: " + masuno);
+  });
+});
 
 /**
  * Rutas para el factor6
