diff --git a/app/nodejs/server.js b/app/nodejs/server.js
index efb73cb..4547631 100755
--- a/app/nodejs/server.js
+++ b/app/nodejs/server.js
@@ -8,6 +8,15 @@ const express = require("express");
 const session = require("express-session");
 const path = require("path");
 const redis = require("redis");
+const winston = require('winston');
+
+const error_logger = winston.createLogger({
+  level: 'error',
+  format: winston.format.json(),
+  transports: [
+    new winston.transports.File({ filename: 'error.log'})
+  ]
+});
 
 /**
  * External services configuration
@@ -88,6 +97,15 @@ app.get("/quiensoy", (req, res) => {
 * Rutas para el factor11
 */
 
+app.get("/dameun500", (req, res) => {
+  throw new Error('Toma un 500!');
+});
+
+app.use(function(err, req, res, next) {
+  error_logger.error(err.stack);
+  res.status(500).send('Toma 500!');
+});
+
 /**
  * Server Activation
  */
