diff --git a/app/dotnet/Controllers/Server.cs b/app/dotnet/Controllers/Server.cs
index d257c1b..014e7e1 100644
--- a/app/dotnet/Controllers/Server.cs
+++ b/app/dotnet/Controllers/Server.cs
@@ -63,7 +63,13 @@ namespace dotnet.Controllers
         ///
         /// Test 11
         ///
-
+        [HttpGet]
+        [Route("dameun500")]
+        public IActionResult dameun500()
+        {
+            throw new Exception("Toma un 500!");
+        }
+        
         ///
         /// Test 9
         /// 
diff --git a/app/dotnet/Startup.cs b/app/dotnet/Startup.cs
index bc69525..0ebe03e 100644
--- a/app/dotnet/Startup.cs
+++ b/app/dotnet/Startup.cs
@@ -5,6 +5,12 @@ using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Caching.Memory;
 using ServiceStack.Redis;
+using Serilog;
+using Serilog.Events;
+using Serilog.Formatting.Compact;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 
 namespace dotnet
 {
@@ -27,6 +33,13 @@ namespace dotnet
             services.AddDistributedMemoryCache();
             services.AddSession();
 
+            services.AddLogging(conf => {
+               conf.ClearProviders();
+               conf.AddSerilog(new LoggerConfiguration()
+                    .WriteTo.File(new CompactJsonFormatter(), "error.log", LogEventLevel.Error)
+                    .CreateLogger());
+            });
+
             services.AddSingleton<RedisClient>(_ => new RedisClient("redis", 6379, "foobared"));
         }
 
@@ -39,7 +52,21 @@ namespace dotnet
             {
                 app.UseDeveloperExceptionPage();
             }
+            else
+            {
+                app.UseExceptionHandler(errorHandler => {
+                    errorHandler.Run(async context => {
+                        var ex = context.Features.Get<IExceptionHandlerPathFeature>();
+                        ILogger<object> logger = context.RequestServices.GetService<ILogger<object>>();
 
+                        logger?.LogError(ex?.Error?.Message);
+                        context.Response.StatusCode = 500;
+                        await context.Response.WriteAsync($"Url: {context.Request.Path}\n");;
+                        await context.Response.WriteAsync($"Error: {ex?.Error?.Message}\n");
+                    });
+                });
+            }
+            
             app.UseSession();
 
             app.UseRouting();
diff --git a/app/dotnet/dotnet.csproj b/app/dotnet/dotnet.csproj
index 103ed39..416261a 100644
--- a/app/dotnet/dotnet.csproj
+++ b/app/dotnet/dotnet.csproj
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Serilog.AspNetCore" Version="3.2.0" />
     <PackageReference Include="ServiceStack.Redis" Version="5.8.0" />
   </ItemGroup>
 
