diff --git a/app/dotnet/Controllers/Server.cs b/app/dotnet/Controllers/Server.cs
index af133b6..f07b331 100644
--- a/app/dotnet/Controllers/Server.cs
+++ b/app/dotnet/Controllers/Server.cs
@@ -1,7 +1,9 @@
+using System;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Configuration;
 using Microsoft.AspNetCore.Http;
+using ServiceStack.Redis;
 
 namespace dotnet.Controllers
 {
@@ -28,6 +30,17 @@ namespace dotnet.Controllers
         ///
         /// Test 4
         ///
+        [HttpGet]
+        [Route("masuno")]
+        public IActionResult masuno([FromServices] RedisClient redisClient)
+        {
+            int masuno = 0;
+            Int32.TryParse(redisClient.Get<string>("masuno"), out masuno);
+            masuno++;
+            redisClient.Set<string>("masuno", masuno.ToString());
+
+            return Ok($"Contador: {masuno}");
+        }
 
         ///
         /// Test 6
diff --git a/app/dotnet/Startup.cs b/app/dotnet/Startup.cs
index 17bff2c..a806c87 100644
--- a/app/dotnet/Startup.cs
+++ b/app/dotnet/Startup.cs
@@ -4,6 +4,7 @@ using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Caching.Memory;
+using ServiceStack.Redis;
 
 namespace dotnet
 {
@@ -26,6 +27,7 @@ namespace dotnet
             ///
             /// Test 4
             ///
+            services.AddSingleton<RedisClient>(_ => new RedisClient("redis", 6379, "foobared"));
             ///
             /// Test 4
             ///
@@ -74,7 +76,7 @@ namespace dotnet
             ///
             /// Test 6
             ///
-            
+
             app.UseRouting();
 
             app.UseEndpoints(endpoints =>
diff --git a/app/dotnet/dotnet.csproj b/app/dotnet/dotnet.csproj
index 2209443..103ed39 100644
--- a/app/dotnet/dotnet.csproj
+++ b/app/dotnet/dotnet.csproj
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="ServiceStack.Redis" Version="5.8.0" />
   </ItemGroup>
 
 </Project>
