diff --git a/app/dotnet/Controllers/Server.cs b/app/dotnet/Controllers/Server.cs
index f07b331..d257c1b 100644
--- a/app/dotnet/Controllers/Server.cs
+++ b/app/dotnet/Controllers/Server.cs
@@ -45,7 +45,21 @@ namespace dotnet.Controllers
         ///
         /// Test 6
         ///
+        [HttpGet]
+        [Route("/diminombre")]
+        public IActionResult diminombre(string nombre)
+        {
+            HttpContext.Session.SetString("user_name", nombre);
+            return Ok($"Hola {nombre}, encantado de conocerte");
+        }
 
+        [HttpGet]
+        [Route("/quiensoy")]
+        public IActionResult quiensoy()
+        {
+            return Ok(HttpContext.Session.GetString("user_name"));
+        }
+        
         ///
         /// Test 11
         ///
diff --git a/app/dotnet/Startup.cs b/app/dotnet/Startup.cs
index cbc2a3e..bc69525 100644
--- a/app/dotnet/Startup.cs
+++ b/app/dotnet/Startup.cs
@@ -24,6 +24,9 @@ namespace dotnet
 
             services.AddMemoryCache();
 
+            services.AddDistributedMemoryCache();
+            services.AddSession();
+
             services.AddSingleton<RedisClient>(_ => new RedisClient("redis", 6379, "foobared"));
         }
 
@@ -37,6 +40,8 @@ namespace dotnet
                 app.UseDeveloperExceptionPage();
             }
 
+            app.UseSession();
+
             app.UseRouting();
 
             app.UseEndpoints(endpoints =>
